### PR TITLE
Add platform argument to getDevices call

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
https://github.com/appium/appium/issues/11892

### CODE REVIEW

File: lib/simctl.js [hunk 1]
```diff
@@ -523,38 +524,39 @@ async function getDevices (forSdk = null) {
      *   }
      * }
      */
-    devices = {};
-    for (let [sdkName, entries] of _.toPairs(JSON.parse(stdout).devices)) {
-      if (sdkName.indexOf('iOS') !== 0) {
+    for (const [sdkName, entries] of _.toPairs(JSON.parse(stdout).devices)) {
+      const versionMatch = new RegExp(`^${_.escapeRegExp(platform)}\\s+(\\S+)`, 'i').match(sdkName);
```
Comment (imurchie) [inline lib/simctl.js]: No need to create the same regular expression each time through the loop.

File: lib/simctl.js [hunk 2]
```diff
@@ -523,38 +524,39 @@ async function getDevices (forSdk = null) {
...
+          ...el,
+          sdk: versionMatch[1]
+        };
       });
     }
   } catch (err) {
     log.debug(`Unable to get JSON device list: ${err.message}`);
-    log.debug('Falling back to manually parsing');
+    log.debug('Falling back to manual parsing');
     devices = await getDevicesByParsing();
```
Comment (imurchie) [inline lib/simctl.js]: So nothing but iOS if we get here?

**Comment by jlipps** [comment]
hmm, seems like #72 should be preferred since it is a bit more general?

File: lib/simctl.js (hunk 1 above)
Comment (mykola-mokhnach) [reply lib/simctl.js]: yep

File: lib/simctl.js (hunk 2 above)
Comment (simongranger) [reply lib/simctl.js]: You should be able to parse for other platforms as well.

Comment (simongranger) [reply lib/simctl.js]: Never mind. Just saw it was updated.

**Comment by simongranger** [comment]
Line 9 seems to still have a reference to iOS. For some reason I cannot add a comment there...

**Comment by simongranger** [comment]
I am assuming we'll be making changes in appium-ios-simulator to allow to pass a platform parameter for getSimulator().

I am using the following code:
`import { getSimulator } from 'appium-ios-simulator';
[...]
this.sim = await getSimulator(this.udid);`

**Comment by mykola-mokhnach** [comment]
@simongranger fixed hardcode at line 9

**Pull request merged**